Content.Server/GameObjects/Components/Kitchen/ReagentGrinderComponent.cs

```diff
@@ -0,0 +1,367 @@
+                return;
+            if (hands.CanPutInHand(item))
+                hands.PutInHand(item);
+
+            _heldBeaker = null;
+            _dirty = true;
+            SetAppearance(ReagentGrinderVisualState.NoBeaker);
+        }
+
+        void IActivate.Activate(ActivateEventArgs eventArgs)
+        {
+            if (!eventArgs.User.TryGetComponent(out IActorComponent? actor) || !Powered)
+            {
+                return;
+            }
+
+            _dirty = true;
+            UserInterface?.Toggle(actor.playerSession);
+        }
+
+        public async Task<bool> InteractUsing(InteractUsingEventArgs eventArgs)
+        {
+            if (!Powered)
+                return false;
+
+            if (!eventArgs.User.TryGetComponent(out IHandsComponent? hands))
+            {
+                Owner.PopupMessage(eventArgs.User, Loc.GetString("You have no hands."));
+                return true;
+            }
+
+            
+            var heldEnt = eventArgs.Using;
+
+            //First, check if user is trying to insert a beaker.
+            //No promise it will be a beaker right now, but whatever.
+            //Maybe this should whitelist "beaker" in the prototype id of heldEnt?
+            if(heldEnt!.TryGetComponent(out SolutionContainerComponent? beaker) && heldEnt!.Prototype!.ID.ToLower().Contains("beaker"))
+            {
+                _beakerContainer.Insert(heldEnt);
+                _heldBeaker = beaker;
+                _dirty = true;
+                //We are done, return. Insert the beaker and exit!
+                SetAppearance(ReagentGrinderVisualState.BeakerAttached);
+                ClickSound();
+                return true;
+            }
+
+            //Next, see if the user is trying to insert something they want to be ground/juiced.
+
+            if(!heldEnt!.TryGetComponent(out GrindableComponent? grind) && !heldEnt!.TryGetComponent(out JuiceableComponent? juice))
+            {
+                //Entity did NOT pass the whitelist for grind/juice.
+                //Wouldn't want the clown grinding up the Captain's ID card now would you?
+                //Why am I asking you? You're biased.
+                return false;
+            }
+
+            //Cap the chamber. Don't want someone putting in 500 entities and ejecting them all at once.
+            //Maybe I should have done that for the microwave too?         
+            if (_chamber.ContainedEntities.Count >= _storageCap)
+            {
+                return false;
+            }
+            _chamber.Insert(heldEnt);
+            _dirty = true;
+
+            return true;
+        }
+
+
+        /// <summary>
+        /// The wzhzhzh of the grinder.
+        /// </summary>
+        /// <param name="isJuiceIntent">true for wanting to juice, false for wanting to grind.</param>
+        private async void DoWork(IEntity user, bool isJuiceIntent)
```

Comment:
Could make the intent an enum instead and handling it would be more graceful I guess but I wouldn't reject a merge over it.

Reply:
hilariously i had a comment in the code that said the exact same thing but i backspaced it